### PR TITLE
Add error codes to site breakage reports

### PR DESCRIPTION
**Pull request by jaceklyp**
Task/Issue URL: https://app.asana.com/0/72649045549333/1205982573174191/f
Tech Design URL: https://app.asana.com/0/0/1206225370438701/f
CC: @jonathanKingston

Description:
Send error and http status codes within reports

Steps to test this PR:

Load https://expired.badssl.com/
Send report
Put a breakpoint and validate that the report contains error
Validate that the report contains httpErrorCodes too

### CODE REVIEW

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -32,8 +32,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/duckduckgo/content-scope-scripts",
       "state" : {
-        "revision" : "38ee7284bac7fa12d822fcaf0677ea3969d15fb1",
-        "version" : "4.59.2"
+        "revision" : "bb027f14bec7fbb1a85d308139e7a66686da160e",
+        "version" : "4.59.0"
```
Comment (federicocappelli) [inline DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: is this downgrade intentional?

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -104,8 +104,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/duckduckgo/privacy-dashboard",
       "state" : {
-        "revision" : "c67d268bf234760f49034a0fe7a6137a1b216b05",
-        "version" : "3.2.0"
+        "branch" : "12-14-feat_macos_allow_direct_navigation_to_breakage_form",
```
Comment (federicocappelli) [inline DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: bad merge or your BSK still have the old privacy dashboard branch?

**Review by federicocappelli — CHANGES_REQUESTED** [review]
something wrong with  "branch" : "12-14-feat_macos_allow_direct_navigation_to_breakage_form",

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (jaceklyp) [reply DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: haven't merged with main yet, should be good now!

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 2 above)
Comment (jaceklyp) [reply DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: same!

**Review by federicocappelli — APPROVED** [review]
LGTM

**Review by federicocappelli — APPROVED** [review]
LGTM, tested and seems fine

**Pull request merged**